src/framework/smoke_queue.hpp

```diff
@@ -0,0 +1,171 @@
+#ifndef SERENITY_SMOKE_QUEUE_HPP
+#define SERENITY_SMOKE_QUEUE_HPP
+
+#include <memory>
+#include <string>
+#include <random>
+#include <vector>
+
+#include <mesos/resources.hpp>
+
+#include <stout/json.hpp>
+#include <stout/option.hpp>
+
+#include "logging/logging.hpp"
+
+#include "smoke_job.hpp"
+
+struct QueueBlock {
+  QueueBlock(std::shared_ptr<SmokeJob> _job,
+              double_t _probability,
+              size_t _index)
+    : baseJob(_job),
+      originalProbability(_probability),
+      baseProbability(_probability),
+      index(_index),
+      aliasJob(nullptr),
+      ready(false) {}
+
+  double_t originalProbability;
+  std::shared_ptr<SmokeJob> baseJob;
+  double_t baseProbability;
+
+  std::shared_ptr<SmokeJob> aliasJob;
+
+  size_t index;
+  bool ready;
+
+  std::string toString() const {
```

Comment:
Could we override std::ostream& operator << (std::ostream &strm, const QueueBlock &a) instead of print method?
Then, we could use LOG(INFO) << <your_instance>